client/pkg/omni/resources/siderolink/pending_machine.go

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package siderolink
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/omni/client/api/omni/specs"
+	"github.com/siderolabs/omni/client/pkg/omni/resources"
+)
+
+// NewPendingMachine creates a new machine being in the limbo state.
+func NewPendingMachine(ns, id string, spec *specs.SiderolinkSpec) *PendingMachine {
```

Comment:
Do we need `ns` as argument? If it always needs to be Ephemeral, we can remove it from the constructor.
Also wondering, why do we have spec as an argument unlike other resources on `SideroLink` and this one?